mobile/library/common/internal_engine.cc

```diff
@@ -272,6 +281,11 @@ envoy_status_t InternalEngine::setPreferredNetwork(NetworkType network) {
   return dispatcher_->post([&, network]() -> void {
     envoy_netconf_t configuration_key =
         Network::ConnectivityManagerImpl::setPreferredNetwork(network);
+    if (!hasIpV6Connectivity()) {
```

Comment:
these few lines are the same as L166 above - perhaps factor out into a little utility function?
Also, a comment here might be helpful to note we must check v6 connectivity before refreshing DNS hosts

Reply:
It's no longer the same since I removed the default `nullopt`. Comment added.